### ISSUE TITLE
hyphens in namespaces result in bad POST

### ISSUE THREAD

**Issue by cisko3000** [miguelgrinberg/Flask-SocketIO#392]
Hi,
I have this in my app:
```

@socketio.on("connect", namespace='/realtime')
def test_connect():
	emit('c response', {'data': 'Connected now'})
```
```
<script type="text/javascript" charset="utf-8">
    var socket = io.connect('http://' + document.domain + ':' + location.port + '/realtime');
    socket.on('connect', function() {
        socket.emit('my event', {data: 'I\'m connected!'});
    });
</script>
```
But when I replace the namespaces in the following lines:
`@socketio.on("connect", namespace='/real-time')`
javascript:
`    var socket = io.connect('http://' + document.domain + ':' + location.port + '/real-time');`

I get bad request when looking at javascript console
http://0.0.0.0:8080/socket.io/ [HTTP/1.1 400 BAD REQUEST 2ms]

Looking at http://socket.io/docs/rooms-and-namespaces/# documentation I see that in javascript, namespaces are allowed hyphens. This makes me think it might be a flask-socketio issue. 

If javascript namespace is hyphenated and python code namespace is not hyphenated, I get the same error in javascript console.

I checked documentation and it says nothing about namespace allowed characters, I guess by namespace they mean a python namespace? because then it would make sense why it doesn't work.

Any thoughts?

**Comment by miguelgrinberg** [comment]
I'll review how the JS client and server handle hyphens. The issue is related to the hyphen also being used as a separator in the encoded Socket.IO packets, so when you put one in a namespace or event name the package parser gets confused.

**Comment by Kurlov** [comment]
Hello,  @miguelgrinberg 
I created PR which might resolve this issue
https://github.com/miguelgrinberg/python-socketio/pull/66

**Comment by miguelgrinberg** [comment]
This issue has been addressed in python-socketio 1.6.3. Please upgrade to that version or newer if you need to use hyphens in namespace or event names.

**Comment by cisko3000** [comment]
Thanks! You guys are awesome
